### PR TITLE
ON-14736 # Added lastUpdatedBy to search for approvals

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: OneBlink.SDK/ApprovalsClient.cs [hunk 1]
```diff
@@ -18,90 +19,70 @@ public ApprovalsClient(string accessKey, string secretKey, TenantName tenantName
...
+            string submittedAfterDateTime = null,
+            string submittedBeforeDateTime = null,
+            List<string> statuses = default(List<string>),
+            string updatedAfterDateTime = null,
+            string updatedBeforeDateTime = null,
+            List<string> lastUpdatedBy = null
+        )
         {
-            string queryString = nameof(limit) + "=" + limit.ToString() + "&" + nameof(offset) + "=" + offset.ToString();
+            NameValueCollection searchParams = System.Web.HttpUtility.ParseQueryString(string.Empty);
```
Comment (kizaonline) [inline OneBlink.SDK/ApprovalsClient.cs:37]:
```suggestion
            NameValueCollection searchParams = NameValueCollection();

```

Comment (mymattcarroll) [reply OneBlink.SDK/ApprovalsClient.cs:37]: Changing to this suggestion causes the querystring to serialised to `/forms-apps/1640/approvals?System.Collections.Specialized.NameValueCollection`

Comment (kizaonline) [reply OneBlink.SDK/ApprovalsClient.cs:37]: Interesting, looks like it relies on a bug https://stackoverflow.com/questions/3865975/namevaluecollection-to-url-query

File: OneBlink.SDK/ApprovalsClient.cs [hunk 2]
```diff
@@ -34,7 +33,7 @@ public ApprovalsClient(string accessKey, string secretKey, TenantName tenantName
             List<string> lastUpdatedBy = null
         )
         {
-            NameValueCollection searchParams = System.Web.HttpUtility.ParseQueryString(string.Empty);
+            IDictionary<string, string> searchParams = new Dictionary<string, string>();
```
Comment (kizaonline) [inline OneBlink.SDK/ApprovalsClient.cs:36]: Will this work for duplicated keys?  e.g. Multiple status's or updatedBy?

Comment (mymattcarroll) [reply OneBlink.SDK/ApprovalsClient.cs:36]: great question, prob not.

**Pull request closed without merging**